### PR TITLE
RHBPMS-4636 - MigrationManager API throws NPE when used in Kie-Spring  environment

### PR DESCRIPTION
**Pull request by tsurdilo**
@mswiderski thanks for all your help on this!

### CODE REVIEW

**Comment by tsurdilo** [comment]
jenkins retest this

File: kie-spring/pom.xml [hunk 1]
```diff
@@ -258,6 +258,7 @@
           <systemPropertyVariables>
             <kie.maven.settings.custom>${project.build.testOutputDirectory}/kie-custom-settings.xml</kie.maven.settings.custom>
           </systemPropertyVariables>
+          <forkMode>always</forkMode>
```
Comment (mswiderski) [inline kie-spring/pom.xml]: @psiroky would it be possible to have a fork for just single test? I am bit afraid this might be too heavy...

Comment (psiroky) [inline kie-spring/pom.xml]: I don't think that's possible using just surefire config.

I agree that `<forkMode>always</forkMode>` is quite heavy. It usually suggests that the tests are not properly isolated or that they are not cleaning-up after themselves.

Comment (mswiderski) [inline kie-spring/pom.xml]: the reason why we have it here is that there are changes in transactionmanagerfactory - change of the impl. So we might be better of to add logic to reset the instance in that https://github.com/droolsjbpm/drools/blob/master/drools-persistence-jpa/src/main/java/org/drools/persistence/TransactionManagerFactory.java

**Comment by tsurdilo** [comment]
@mswiderski amended per your suggestion - thanks. Depends on https://github.com/droolsjbpm/drools/pull/1139

will ask to retests once that pr is merged.

**Comment by mswiderski** [comment]
Nice work @tsurdilo 

I'll leave it open for other to take a look and then merge on Monday

**Comment by mswiderski** [comment]
jenkins retest this

**Pull request merged**